### ISSUE TITLE
Validation set during training and evaluating best model

### ISSUE THREAD

**Issue by CCXD** [deepchem/deepchem#372]
So currently the validation set is not used at all during training. Also, the model of the last-finished epoch is chosen for evaluation, instead of the model corresponding to the epoch with the lowest loss.

Is this something that is desired as a PR? I have a rough implementation of valid loss-based evaluation and early stopping.

**Comment by rbharath** [comment]
This is a valid use case, but a little tricky. We've frozen development on `TensorflowMultitaskClassifier` and `TensorflowMultitaskRegressor` (except bugfixes) for the next month while we try to get a paper out.

One strategy might be to make a subclass that implements validation based early stopping (say `TensorflowValidLossClassifier`). This would let us keep stability on the parent class, and wouldn't affect on-going refactor #369. PR welcome for a subclass :-)

**Comment by joegomes** [comment]
I'm onboard with merging the subclass into the main TF models once we unfreeze. I've also implemented a modification of `fit()` to do early stopping and think this would be a very useful contribution.

**Comment by rbharath** [comment]
This might be a useful addition, but the specifics in the discussion above are out of date. Please feel free to open a new issue discussing feature additions to TensorGraph if interested.
